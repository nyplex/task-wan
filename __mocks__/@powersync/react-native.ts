module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
