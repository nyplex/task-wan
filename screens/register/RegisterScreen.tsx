import { Box } from "@/components/gluestack/box";
import { HStack } from "@/components/gluestack/hstack";
import { CLASS_NAMES } from "@/constants/CLASS_NAMES";
import Text from "@/components/Text";
import Divider from "@/components/layout/Divider";
import Header from "./components/Header";
import RegisterForm from "./components/RegisterForm";
import SocialAuthButton from "@/components/buttons/SocialAuthButton";

const RegisterScreen = () => {
  return (
    <Box className={CLASS_NAMES.DEFAULT_SCREEN_WRAPPER}>
      <Header />
      <Text weight="semi-bold" className="text-center">
        Create your account
      </Text>
      <RegisterForm />
      <Box className="mt-12 items-center">
        <Divider title="Or register with" width="80%" />
        <HStack className="items-center justify-around mt-12 w-full">
          <SocialAuthButton provider="google" />
          <SocialAuthButton provider="apple" />
        </HStack>
      </Box>
    </Box>
  );
};

export default RegisterScreen;
