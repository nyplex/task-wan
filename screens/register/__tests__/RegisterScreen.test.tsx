import { render } from "@testing-library/react-native";
import RegisterScreen from "../RegisterScreen";

// Mock all child components to isolate RegisterScreen
const Header = () => <></>;
Header.displayName = "Header";
const RegisterForm = () => <></>;
RegisterForm.displayName = "RegisterForm";
const Divider = () => <></>;
Divider.displayName = "Divider";
const SocialAuthButton = () => <></>;
SocialAuthButton.displayName = "SocialAuthButton";

jest.mock("../components/Header", () => Header);
jest.mock("../components/RegisterForm", () => RegisterForm);
jest.mock("@/components/layout/Divider", () => Divider);
jest.mock("@/components/buttons/SocialAuthButton", () => SocialAuthButton);

describe("RegisterScreen", () => {
  it("renders without crashing", () => {
    render(<RegisterScreen />);
    // No error thrown means pass
  });

  it("renders Header, RegisterForm, Divider, and SocialAuthButtons", () => {
    const { UNSAFE_queryAllByType } = render(<RegisterScreen />);
    expect(UNSAFE_queryAllByType(Header).length).toBe(1);
    expect(UNSAFE_queryAllByType(RegisterForm).length).toBe(1);
    expect(UNSAFE_queryAllByType(Divider).length).toBe(1);
    expect(UNSAFE_queryAllByType(SocialAuthButton).length).toBe(2);
  });

  it("renders the register prompt text", () => {
    const { getByText } = render(<RegisterScreen />);
    expect(getByText("Create your account")).toBeTruthy();
  });

  it("matches the rendered output snapshot", () => {
    const { toJSON } = render(<RegisterScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
