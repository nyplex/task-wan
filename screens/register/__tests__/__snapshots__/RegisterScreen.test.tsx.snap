// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen matches the rendered output snapshot 1`] = `
<View
  className="flex-1 bg-backrgound py-safe-offset-4 px-4"
>
  <Text
    className="font-body based-styles font-semibold text-typography-primary text-[16px] tracking-[0px] text-center"
  >
    Create your account
  </Text>
  <View
    className="mt-12 items-center"
  >
    <View
      className="flex-row items-center justify-around mt-12 w-full"
    />
  </View>
</View>
`;
